src/main/java/com/google/api/gax/grpc/HeaderInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package com.google.api.gax.grpc;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall.SimpleForwardingClientCall;
+import io.grpc.ForwardingClientCallListener.SimpleForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+
+/**
+ * An intercepter to handle custom header.
+ */
+public class HeaderInterceptor implements ClientInterceptor {
```

Comment:
I have added a unit test. I used this https://cs.corp.google.com/piper///depot/google3/third_party/java_src/grpc/core/javatests/io/grpc/ClientInterceptorsTest.java as a reference.